lib/cisco_node_utils/cmd_ref/README_YAML.md

```diff
@@ -0,0 +1,457 @@
+```
+
+### `config_get_token_append`
+
+When using a `_template` section, an attribute can use
+`config_get_token_append` to extend the `config_get_token` value provided by
+the template instead of replacing it:
+
+```yaml
+# interface.yaml
+_template:
+  config_get: 'show running-config interface all'
+  config_get_token: '/^interface <name>$/i'
+
+description:
+  config_get_token_append: '/^description (.*)$/'
+  # config_get_token value for 'description' is now:
+  # ['/^interface <name>$/i', '/^description (.*)$/']
+```
+
+This can also be used to specify conditional tokens which may or may not be
+used depending on the set of parameters passed into `config_get()`:
+
+```yaml
+# ospf.yaml
+_template:
+  config_get: 'show running ospf all'
+  config_get_token: '/^router ospf <name>$/'
+  config_get_token_append:
+    - '/^vrf <vrf>$/'
+
+router_id:
+  config_get_token_append: '/^router-id (\S+)$/'
+```
+
+In this example, the `vrf` parameter is optional and a different
+`config_get_token` value will be generated depending on its presence or absence:
+
+```ruby
+irb(main):008:0> ref = cr.lookup('ospf', 'router_id')
+irb(main):012:0> ref.config_get_token(name: 'red')
+=> [/^router ospf red$/, /^router-id (\S+)?$/]
+irb(main):013:0> ref.config_get_token(name: 'red', vrf: 'blue')
+=> [/^router ospf red$/, /^vrf blue$/, /^router-id (\S+)?$/]
+```
+
+### `config_set`
+
+The `config_set` parameter is a string or array of strings representing the
+configuration CLI command(s) used to set the value of the attribute.
+
+```yaml
+# interface.yaml
+create:
+  config_set: 'interface <name>'
+
+description:
+  config_set: ['interface <name>', 'description <desc>']
+```
+
+### `config_set_append`
+
+When using a `_template` section, an attribute can use `config_set_append` to
+extend the `config_set` value provided by the template instead of replacing it:
+
+```yaml
+# interface.yaml
+_template:
+  config_set: 'interface <name>'
+
+access_vlan:
+  config_set_append: 'switchport access vlan <number>'
+  # config_set value for 'access_vlan' is now:
+  # ['interface <name>', 'switchport access vlan <number>']
+```
+
+Much like `config_get_token_append`, this can also be used to specify optional
+commands that can be included or omitted as needed:
+
+```yaml
+# ospf.yaml
+_template:
+  config_set: 'router ospf <name>'
+  config_set_append:
+    - 'vrf <vrf>'
+
+router_id:
+  config_set_append: 'router-id <router_id>'
+```
+
+```ruby
+irb(main):008:0> ref = cr.lookup('ospf', 'router_id')
+irb(main):017:0> ref.config_set(name: 'red', state: nil, router_id: '1.1.1.1')
+=> ["router ospf red", " router-id 1.1.1.1"]
+irb(main):019:0> ref.config_set(name: 'red', vrf: 'blue',
+                                state: 'no', router_id: '1.1.1.1')
+=> ["router ospf red", "vrf blue", "no router-id 1.1.1.1"]
+```
+
+### `default_value`
+
+If there is a default value for this attribute when not otherwise specified by
```

Comment:
Rewrote this section:

### `default_value`

If there is a default value for this attribute when not otherwise specified by
the user, the `default_value` parameter describes it. This can be a string,
boolean, integer, array, or nil.

```yaml
description:
  default_value: ''

hello_interval:
  default_value: 10

auto_cost:
  default_value: [40, 'Gbps']

ipv4_address:
  # YAML represents nil as ~
  default_value: ~
```

By convention, a `default_value` of `''` (empty string) represents a configurable property that defaults to absent, while a default of `nil` (Ruby) or `~` (YAML) represents a property that has no meaningful default at all.